<a href="https://colab.research.google.com/github/felstuff/BackToBasic/blob/master/Titanic_Solution_with_MinSuk_Heo.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Titanic : Machine Learning fr Disaster 

**Predict survival on the Titanic** 


*   Defining the problem statement 
*   Collecting the data 
*   Exploratory data analysis
*   Feature engineering
*   Feature selection
*   Modelling
*   Testing











**Load Train, and Test dataset using Pandas**



In [8]:
import pandas as pd

train = pd.read_csv('train.csv')
test= pd.read_csv('test.csv')
test.head()

#used to import data, hence everything else you used to trained after have to be pandas  

PassengerId  Pclass  ... Cabin Embarked
0          892       3  ...   NaN        Q
1          893       3  ...   NaN        S
2          894       2  ...   NaN        Q
3          895       3  ...   NaN        S
4          896       3  ...   NaN        S

[5 rows x 11 columns]

**Data Dictionary**

*   Survived: 0 = No; 1 = Yes
*   pclass: Ticket Class 1=1st; 2=2nd; 3=3rd 
*   Sibsp: # of sibiling/ spouse on board of Titanic 
*   parch: # of parents/ children on board of Titanic 
*   Ticket: Ticket Number 
*   Cabin: Cabin Number 
*   Embarked: Port of entry C = Cherbourg; Q = Queenstown, S = Southampton 



**Total rows and columns**

We can see that there are 892 rows and 12 columns in our training dataset 


In [9]:
test.head()

PassengerId  Pclass  ... Cabin Embarked
0          892       3  ...   NaN        Q
1          893       3  ...   NaN        S
2          894       2  ...   NaN        Q
3          895       3  ...   NaN        S
4          896       3  ...   NaN        S

[5 rows x 11 columns]

In [10]:
train.shape
#this is only for pandas (pd) library, to tell you how many row and columns in the train dataset  

(891, 12)

In [11]:
test.shape

(418, 11)

In [21]:
train.info()
#if without the "()" it will show you the list of data on the csv

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [22]:
test.info ()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  418 non-null    int64  
 1   Pclass       418 non-null    int64  
 2   Name         418 non-null    object 
 3   Sex          418 non-null    object 
 4   Age          332 non-null    float64
 5   SibSp        418 non-null    int64  
 6   Parch        418 non-null    int64  
 7   Ticket       418 non-null    object 
 8   Fare         417 non-null    float64
 9   Cabin        91 non-null     object 
 10  Embarked     418 non-null    object 
dtypes: float64(2), int64(4), object(5)
memory usage: 36.0+ KB


In [27]:
train.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [28]:
test.isnull().sum()

PassengerId      0
Pclass           0
Name             0
Sex              0
Age             86
SibSp            0
Parch            0
Ticket           0
Fare             1
Cabin          327
Embarked         0
dtype: int64